lib/box/client.go

```diff
@@ -135,8 +136,19 @@ func (err ExitError) Error() string {
 func dial(socketPath string) (net.Conn, error) {
 	conn, err := net.Dial("unix", socketPath)
 	if err != nil {
```

Comment:
@a-palchikov I can't see an easy Go-like way to check whether the error returned by net.Dial contains "permission denied" or not - do you have any advice?

Reply:
`net.Dial` wraps the original error into `net.OpError`, so you could use a function similar to this to check whether the error indicates a permission issue:
```go
func dial(socketPath string) {
	conn, err := net.Dial("unix", socketPath)
	if err == nil {
		return conn, nil
	}
	if isPermissionDenied(err) {
		return nil, trace.Wrap(err, "permission denied accessing planet socket. Try with `sudo`?")
	}
	return nil, checkError(trace.Wrap(err, "failed to connect to planet socket"))
}

func isPermissionDenied(err error) bool {
	if opErr, ok := trace.Unwrap(err).(*net.OpError); ok {
		return opErr.Err != nil && os.IsPermission(opErr.Err)
	}
	return os.IsPermission(trace.Unwrap(err))
}
```

Reply:
Thanks!